tests/services/test_links.py

```diff
@@ -10,52 +10,51 @@
 """Test links."""
 
 import pytest
-from flask_principal import Identity
+from marshmallow import Schema, fields
 
 
-@pytest.fixture()
-def identity_no_need():
-    """Simple identity fixture without needs."""
-    i = Identity(1)
-    return i
+class MockResourceRequestCtx:
 
+    url_args = {"q": ""}
 
-@pytest.mark.skip()
-def test_links(service, identity_simple, example_record, es):
-    """Test record links creation."""
-    pid_value = str(example_record.id)
-
-    # NOTE: We are testing linker.links() as opposed to record_unit.links
-    #       because the former is used in all RecordService methods (not just
-    #       create)
-    links = service.linker.links(
-        "record",
-        identity_simple,
-        pid_value=pid_value,
-        record=example_record,
-    )
-
-    expected_links = {
-        "self": f"https://localhost:5000/api/records/{pid_value}",
-        "delete": f"https://localhost:5000/api/records/{pid_value}",
-        # NOTE: Generate the link even if no file(s) for now
-        "files": f"https://localhost:5000/api/records/{pid_value}/files",
-    }
 
-    assert expected_links == links
+class MockRecordLinksSchema(Schema):
+    """Schema for a record's links."""
+
+    self = fields.String(default="/api/records/12345-ABCDE")
+
+
+class MockSearchLinksSchema(Schema):
+    """Schema for a search result's links."""
+
+    self = fields.String(default="/api/records?q=")
 
 
-# TODO: example record needs to be permission aware
-@pytest.mark.skip()
-def test_links_with_permissions(service, identity_no_need, example_record, es):
-    """Test record links creation with permissions."""
-    pid_value = str(example_record.id)
```

Comment:
We should keep this skipped test around: we haven't tested that permission can regulate link generation with the new way of generating links.